firmware/cmake/tests.cmake

```diff
@@ -23,7 +23,6 @@ function(compile_gtest_executable
     target_compile_options(${TEST_EXECUTABLE_NAME}
         PUBLIC
         -Wall
-        -g3
```

Comment:
Any reason why we drop `-g3`?

Reply:
redundant because it is specified with CMAKE_BUILD_TYPE.

the idea is to support as many toolchains as possible. -g3 is gcc flag only, /g3 for msvc for example.